plugin/src/main/resources/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition/config.jelly

```diff
@@ -28,7 +28,7 @@
   <f:entry title="${%Script}" field="script">
     <wfe:workflow-editor />
   </f:entry>
-  <f:entry field="sandbox">
+  <f:entry field="sandbox" class="${instance.shouldHideSandbox() ? 'jenkins-hidden' : ''}">
```

Comment:
On a newly created project, `instance == null`, so this should be a method on the `descriptor` passing the `instance` as a `@CheckForNull` arg. Otherwise JEXL will silently convert the implicit NPE to a false and show the checkbox (I think).

Reply:
Good catch! Fixed.